### PR TITLE
Fix: filter issue with skipped filter effects

### PR DESCRIPTION
**Pull request by GoodBoyDigital**
added some tests

### CODE REVIEW

File: tests/scene/filter-effect.tests.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    {
+        const container = new Container();
+
+        expect(container.effects.length).toBe(0);
+
+        expect(container.filters).toBe(undefined);
+
+        container.filters = [];
+
+        expect(container.effects.length).toBe(1);
```
Comment (Zyie) [inline tests/scene/filter-effect.tests.ts:18]: why would filters have a length of 1 here after being set to an empty array?

File: tests/scene/filter-effect.tests.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    it('should set filter effects correctly if a modifed array is passed in', async () =>
+    {
+        const container = new Container();
+
+        const filters: Filter[] = [];
+
+        container.filters = filters;
+
+        expect(container.effects.length).toBe(1);
```
Comment (Zyie) [inline tests/scene/filter-effect.tests.ts:63]: same here?

**Comment by GoodBoyDigital** [comment]
thanks for the sanity check!! updated the PR - works better now :)

File: tests/visual/scenes/container/skip-filter.scene.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import { ColorMatrixFilter } from '../../../../src/filters/defaults/color-matrix/ColorMatrixFilter';
+import { NoiseFilter } from '../../../../src/filters/defaults/noise/NoiseFilter';
+import { Container } from '../../../../src/scene/container/Container';
+import { Graphics } from '../../../../src/scene/graphics/shared/Graphics';
+
+import type { TestScene } from '../../types';
+
+export const scene: TestScene = {
+    it: 'should skip filters correctly',
+    only: true,
```
Comment (Zyie) [inline tests/visual/scenes/container/skip-filter.scene.ts:10]:
```suggestion
```

**Review by Zyie — APPROVED** [review]

**Pull request merged**